OutOfSchool/OutOfSchool.WebApi.Tests/Extensions/ApplicationExtensionTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Tests.Common.TestDataGenerators;
+using OutOfSchool.WebApi.Extensions;
+
+namespace OutOfSchool.WebApi.Tests.Extensions;
+
+[TestFixture]
+public class ApplicationExtensionTests
+{
+    [Test]
+    public void ApplicationExtensions_AmountOfPendingApplications_ReturnValue()
+    {
+        // Arrange
+        var applications = GetApplications();
+
+        // Act
+        var result = applications.AmountOfPendingApplications();
+        var expected = applications.Count(x =>
+            x.Status == ApplicationStatus.Pending
+            && !x.IsDeleted
+            && (x.Child == null || !x.Child.IsDeleted)
+            && (x.Parent == null || !x.Parent.IsDeleted));
```

Comment:
(Fragile tests, duplicated code) Please avoid testing method's code by writing the same code in test. This way you are not checking the requirements, but rather your understanding of them (and if you had taken/coded them wrong once - you wouldn't have got any chance to catch this by unit-test). 

Instead check the requirements. Separately. Because it's a `unit`-test. Not a `method`-test.

Assert, that `deleted` applications doesn't affect the count (1 test), applications in other status than `Pending` doesn't affect (another test), check how `not-`/`null` `not-`/`deleted` `child` affects the count, same for `parent`. 
(Yes, it would be many, but pretty simple (dumb) tests.) 

So, in each test you could start with a list of ideal applications, make N applications "wrong" due to one of the requirement (or their combination) and ensure that count returned by `AmountOfPendingApplications()` is changed by N.

Summing up: the idea is that your test`s code should be simplier than code of your logic, thus you could do less mistakes in unit-test, it is easier to maintain, failed test has only one reason to fail ... y'know, all this FIRST principles.

Reply:
Thanks. Understand. I made short tests for different situations.

Reply:
@VadymLevkovskyi I missed your nick in the comment before.  Sorry.

Reply:
Np, you don't need to mention me. I get notifications on any changes in PR where I'm set as a reviewer.
And thanks for redesign!